src/sentry/api/endpoints/group_integration_details.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import
+
+from django.db import IntegrityError, transaction
+
+from rest_framework.response import Response
+
+from sentry.api.bases import GroupEndpoint
+from sentry.models import ExternalIssue, GroupLink, OrganizationIntegration
+
+
+class GroupIntegrationDetails(GroupEndpoint):
+    # was thinking put for link an existing issue, post for create new issue?
+    def put(self, request, group, integration_id):
+        organization_id = group.project.organization_id
+        try:
+            # check org permissions
+            # TODO(jess): should this eventually check ProjectIntegration?
+            integration = OrganizationIntegration.objects.filter(
+                integration_id=integration_id,
+                organization_id=organization_id,
+            ).select_related('integration').get().integration
+        except OrganizationIntegration.DoesNotExist:
+            return Response(status=404)
+
+        external_issue_id = request.DATA.get('externalIssue')
+
+        if not external_issue_id:
+            return Response({'detail': 'External ID required'}, status=400)
```

Comment:
Should you just check this first to avoid the DB queries in this case?

Reply:
I mean, this is just validating the post data is the right shape.

I don't have a strong opinion here, so if you don't think it's right, that's fine.